DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -27,1094 +27,614 @@
-				p_HighResMesh=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--low-res-meshes=*)
-				p_LowResMeshes=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--registration-name=*)
-				p_RegName=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--dedrift-reg-files=*)
-				p_DeDriftRegFiles=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--concat-reg-name=*)
-				p_ConcatRegName=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--maps=*)
-				p_Maps=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--myelin-maps=*)
-				p_MyelinMaps=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-concat-names=*)
-				p_mrFIXConcatNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-names=*)
-				p_mrFIXNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-extract-concat-names=*)
-				p_mrFIXExtractConcatNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-extract-names=*)
-				p_mrFIXExtractNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-extract-extra-regnames=*)
-				p_mrFIXExtractExtraRegNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--multirun-fix-extract-volume=*)
-				p_mrFIXExtractDoVol=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--fix-names=* | --rfmri-names=*)
-				p_fixNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--dont-fix-names=* | --tfmri-names=*)
-				p_dontFixNames=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--smoothing-fwhm=*)
-				p_SmoothingFWHM=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--high-pass=* | --highpass=*)
-				p_HighPass=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--myelin-target-file=*)
-				p_MyelinTargetFile=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--input-reg-name=*)
-				p_InRegName=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--matlab-run-mode=*)
-				p_MatlabRunMode=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			--motion-regression=*)
-				p_MotionRegression=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			*)
-				show_usage
-				log_Err_Abort "unrecognized option: ${argument}"
-				;;
-		esac
-	done
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
+# Show wb_command version
+log_Msg "Showing Connectome Workbench (wb_command) version"
+${CARET7DIR}/wb_command -version
 
-	local error_count=0
+#display the parsed/default values
+opts_ShowValues
 
-	# check required parameters
-	if [ -z "${p_StudyFolder}" ]; then
-		log_Err "Study Folder (--path= or --study-folder=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "Study Folder: ${p_StudyFolder}"
+if [[ "${OutputRegName}" == "" ]]; then
+	OutputRegName="${RegName}"
+else
+	if [[ "${DeDriftRegFiles}" == "" ]]; then
+		log_Err "--concat-reg-name must not be used unless dedrifting (just remove --concat-reg-name=<whatever> from your command)"
```

Comment:
This should probably be `log_Err_Abort`.  `log_Err` continues without exiting, which is generally not what should happen after an error.

Reply:
The reason the previous parsing code used `log_Err` was to report all the issues with the arguments before exiting (hence the error count variable).  The new parsing does that internally for the mandatory arguments, but I think it would be unnecessary complication to implement it for the additional "these can't be used together" or other detailed sanity checking logic.